packages/kbn-health-gateway-server/src/kibana/kibana_config.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { readFileSync } from 'fs';
+import type { Duration } from 'moment';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { ServiceConfigDescriptor } from '@kbn/core-base-server-internal';
+
+const hostURISchema = schema.uri({ scheme: ['http', 'https'] });
+
+const configSchema = schema.object({
```

Comment:
I tested with an empty config file and still saw `[server] Server running on http://localhost:3000` whereas I would expect to get a configuration validation exception

When I ran the server with an invalid config (deleted `kibana.hosts`) it seemed to correctly start up:
```
[2022-10-21T03:47:45.496+02:00][DEBUG][config] Marking config path as handled: logging
[2022-10-21T03:47:45.496+02:00][DEBUG][config] Marking config path as handled: kibana
[2022-10-21T03:47:45.496+02:00][DEBUG][config] Marking config path as handled: server
[2022-10-21T03:47:45.522+02:00][INFO ][server] Server running on https://localhost:3000
[2022-10-21T03:47:45.523+02:00][DEBUG][server] registering route handler for [/api/status]
```

Reply:
This is expected based on the way it is currently written (it includes defaults for all settings), but I agree that in production it would be better to simply fail if no config is provided for the kibana hosts. Will remove those defaults.